src/middleware/indexer.js

```diff
@@ -162,18 +174,67 @@ const findLikelyNFTs = async (ctx) => {
             and action_kind = 'FUNCTION_CALL'
             and args->>'args_json' is not null
             and args->>'method_name' like 'nft_%'
+            and receipt_included_in_block_timestamp <= $2
+            and receipt_included_in_block_timestamp > $3
     `;
 
     const ownershipChangeEvents = `
         select distinct emitted_by_contract_account_id as receiver_account_id 
         from assets__non_fungible_token_events
         where token_new_owner_account_id = $1
+            and emitted_at_block_timestamp <= $2
+            and emitted_at_block_timestamp > $3
     `;
 
-    const { rows } = await pool.query([ownershipChangeFunctionCalls, ownershipChangeEvents].join(' union '), [accountId]);
+    const { rows } = await pool.query([ownershipChangeFunctionCalls, ownershipChangeEvents].join(' union '), [accountId, lastBlockTimestamp, fromBlockTimestamp]);
+    return { rows, lastBlockTimestamp };
+};
+
+const findLikelyTokens = async (ctx) => {
+    const { accountId } = ctx.params;
+
+    const { rows } = await likelyTokensFromBlock(0, accountId);
     ctx.body = rows.map(({ receiver_account_id }) => receiver_account_id);
```

Comment:
Are there any callers of `likelyTokensFromBlock` or `likelyNFTsFromBlock` that _don't_ need to map this data to return a list of strings? It seems like this `.map` should be happening in the `likely*FromBlock` methods if nothing is relying on `rows` being a list of objects. Individual callers shouldn't be responsible for doing the `.map` if that's the expected shape of the returned value.

Reply:
I'd prefer to leave this as is since this can be generalized to other uses aside from listing `receiver_account_id`s. I did the same thing for `findLastBlockByTimestamp`. It's good to have generic methods that query pg and return rows here imo.

Reply:
Gotcha, good point 👍